test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1050,3 +1051,353 @@ def test_ded_time_queue(self):
+        a = {'Resource_List.ncpus': 2}
+        jid1 = self.submit_jobs(4, a, user=TEST_USER1)
+        jid2 = self.submit_jobs(4, a, user=TEST_USER2)
+        jid3 = self.submit_jobs(4, a, user=TEST_USER3)
+
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'True'})
+
+        # One equivalence class
+        m = self.scheduler.log_match("Number of job equivalence classes: 1",
+                                     max_attempts=10, starttime=self.t)
+        self.assertTrue(m)
+
+        # Alter a queued job
+        self.t = time.time()
+        self.server.alterjob(jid3[2], {ATTR_N: "test"})
+
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'True'})
+
+        # Three equivalence classes: one is for queued jobs that
+        # do not have cput set. 2 for the different cputime value
+        # set by runjob and modifyjob hook
+        m = self.scheduler.log_match("Number of job equivalence classes: 3",
+                                     max_attempts=10,
+                                     starttime=int(self.t) - 1)
+        self.assertTrue(m)
+
+    def test_mom_hook(self):
+        """
+        Test for job equivalence classes with mom hooks.
+        """
+
+        # Create resource
+        attrib = {}
+        attrib['type'] = "string_array"
+        attrib['flag'] = 'h'
+        self.server.manager(MGR_CMD_CREATE, RSC, attrib, id='foo_str')
+
+        # Create vnodes
+        a = {'resources_available.ncpus': 4,
+             'resources_available.foo_str': "foo,bar,buba"}
+        self.server.create_vnodes('vnode', a, 4, self.mom)
+
+        # Add resources to sched_config
+        self.scheduler.add_resource("foo_str")
+
+        # Create execjob_begin hook
+        hook1 = """
+import pbs
+e = pbs.event()
+j = e.job
+
+if j.Resource_List["host"] == "vnode[0]":
+    j.Resource_List["foo_str"] = "foo"
+elif j.Resource_List["host"] == "vnode[1]":
+    j.Resource_List["foo_str"] = "bar"
+else:
+    j.Resource_List["foo_str"] = "buba"
+"""
+
+        a = {'event': "execjob_begin", 'enabled': 'True'}
+        self.server.create_import_hook("test", a, hook1)
+
+        # Turn off the scheduling
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'False'})
+
+        # Submit jobs
+        a = {'Resource_List.select': "vnode=vnode[0]:ncpus=2"}
+        jid1 = self.submit_jobs(2, a)
+        a = {'Resource_List.select': "vnode=vnode[1]:ncpus=2"}
+        jid2 = self.submit_jobs(2, a)
+        a = {'Resource_List.select': "vnode=vnode[2]:ncpus=2"}
+        jid3 = self.submit_jobs(2, a)
+
+        # Turn on the scheduling
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'True'})
+
+        # One equivalence class as execjob_begin hook runs
+        # after job is scheduled by scheduler
+        m = self.scheduler.log_match("Number of job equivalence classes: 3",
```

Comment:
Actually the reason I was expecting one because of the execjob_begin hook. I thought it will be picked up in the next cycle but I was wrong. anyways I have corrected it now.